cms/server/contest/handlers/tasksubmission.py

```diff
@@ -67,9 +67,16 @@ class SubmitHandler(ContestHandler):
     """
 
     @tornado_web.authenticated
-    @actual_phase_required(0, 3)
+    @actual_phase_required(0, 1, 2, 3, 4)
     @multi_contest
     def post(self, task_name):
+        # Reject submission if the contest disallow unofficial submission outside of official window or analysis mode
+        if not self.contest.allow_unofficial_submission_after_participation_ended and \
```

Comment:
Nit: after the above comment, I think the condition is more readable if it is adjusted like this:

```
if 0 < self.r_params["actual_phase"] < 3 and \
    not self.contest.allow_unofficial_submission_after_participation_ended
```

What do you think?